### DISCUSSION TITLE
Set focus to Dropdown elements

### DISCUSSION THREAD

**Discussion by aryanshridhar** [algolia/instantsearch#4606] [Q&A] [answered]
Hey , Is it possible to set focus to the dropdown elements through which the user can navigate the list using arrow up/down key.
 
Similar issue is often occurred in <a href = 'https://www.electronjs.org/'>electronjs</a> website (which uses instantsearch) where users are unable to navigate through the list (after searching) using up/down arrow keys.

**Comment by eunjae-lee** [comment, marked as answer]
Hello @aryanshridhar 
Do you mean a dropdown like this?
![image](https://user-images.githubusercontent.com/499898/102195175-8621c300-3ebe-11eb-93f5-0ccbe6672a71.png)

If you're already using InstantSearch for such UI, it'd be helpful to know how you have implemented it (like a little bit of your code snippet).
If you are starting to implement something new, then [autocomplete](https://autocomplete.algolia.com/) can be a good fit for you. You can also mix it with InstantSearch, too.

It really depends on what you're building. Let me know what your case is :)

**Reply by aryanshridhar** [reply]
Yeah @eunjae-lee , As of now the user can't navigate through the above dropdown items using arrow keys . 

<strong>Expected :</strong>

![Peek 2020-12-15 15-24](https://user-images.githubusercontent.com/53977614/102199670-e37f3980-3ee9-11eb-9706-4f6b597d6a4d.gif)

Is it possible to handle keyboard focus events with it ?

Edit : One way to tackle this scenario , is it to write custom focus event binding it appropriately with dropdown scrollTop/scrollBottom methods within <a href = 'https://www.algolia.com/doc/api-reference/widgets/instantsearch/ios/#widget-param-onresults'><code>onResults</code></a> function .

**Reply by eunjae-lee** [reply]
InstantSearch.js doesn't ship something like that out of the box. You can still add an event listener to your window to capture key events and apply a specific className to an item for highlights.

However as I said before, I recommend you to try out [autocomplete](https://autocomplete.algolia.com/) which has all those experience implemented out of the box.

You can implement dropdown experience like this:
![Kapture 2020-12-15 at 14 55 01](https://user-images.githubusercontent.com/499898/102223869-8a150b80-3ee5-11eb-934a-3431bf33dcb0.gif)

**Reply by aryanshridhar** [reply]
Cool ! Thanks for it :+1: , Will look into <a href = 'https://autocomplete.algolia.com/'>autocomplete</a> .

**Reply by eunjae-lee** [reply]
Let me know if you need any help! :)
